lib/apimatic-core/logger/configurations/api_logging_configuration.rb

```diff
@@ -0,0 +1,112 @@
+module CoreLibrary
+  # Represents options for configuring logging behavior.
+  class LoggingConfiguration
+    attr_accessor :logger, :log_level, :request_logging_config, :response_logging_config, :mask_sensitive_headers
+
+    # Initializes a new instance of LoggingConfiguration.
+    #
+    # @param logger [LoggerInterface] The logger to use for logging messages.
+    # @param log_level [LogLevel] The log level to determine which messages should be logged.
+    # @param request_logging_config [RequestLoggingConfiguration] Options for logging HTTP requests.
+    # @param response_logging_config [ResponseLoggingConfiguration] Options for logging HTTP responses.
+    # @param mask_sensitive_headers [Boolean] Indicates whether sensitive headers should be masked in logged messages.
+    def initialize(
+      logger: ConsoleLogger.new,
```

Comment:
Blocker: instead of setting the default value at parameter level, set the default behavior in the constructor body as following
``` Ruby
@logger = logger.nil? ? ConsoleLogger.new : logger
```